engine/src/platform-window.cpp

```diff
@@ -65,6 +65,9 @@ MCPlatformWindow::MCPlatformWindow(void)
     m_is_realized = false;
 	
 	m_is_opaque = true;
+    
+    // MERG-2015-10-11: [[ DocumentFilename ]] documentFilename property
+    m_document_filename = nil;
```

Comment:
Might it be better to initialise:

````c++
m_document_filename = MCValueRetain(kMCEmptyString);
````

and then require you could remove quite a lot of null checks for `m_document_filename` (i.e. the assumption would be that it must always be a valid `MCStringRef`)?